### ISSUE TITLE
Surface Laptop Go 3

### ISSUE THREAD

**Issue by nickdepinet** [linux-surface/linux-surface#1275]
Hello, I've attached 
[diagnostics.tar.gz](https://github.com/linux-surface/linux-surface/files/13038297/diagnostics.tar.gz)

for the SLG3 from linux mint 21.2 liveusb. 

After install, device bootloops at loading initial ramdisk (regular & recovery mode, secure boot off), until you disable 'secured core' in uefi. this is new behavior to me. After disabling, i was able to boot into linux with the 5.15.0-76-generic kernel that ships with linux mint 21.2. Will install latest surface-kernel and test functionality with that.

**Comment by nickdepinet** [comment]
no battery info on default 5.15 kernel, as expected from SLG2

**Comment by qzed** [comment]
I assume it's the same situation as on the SLG1 and SLG2, meaning battery is handled via SAM. I'll have a more detailed look at it on the weekend, but if you want to try something out in the meantime, you could add an entry for the SLG3 to [this](https://elixir.bootlin.com/linux/v6.5.7/source/drivers/platform/surface/surface_aggregator_registry.c#L321) array. The new device ID should be `MSHW0440` and you can probably just point to the node group used by the SLG1 and SLG2.

**Comment by qzed** [comment]
Oh, and instead of patching the kernel, you can also try adding it to https://github.com/linux-surface/surface-aggregator-module/ ([here](https://github.com/linux-surface/surface-aggregator-module/blob/653b44a6d615f1375434e14ac87b2d2c3a5066e9/module/src/clients/surface_aggregator_registry.c#L321)) and compile that.

**Comment by qzed** [comment]
Acpidump added in https://github.com/linux-surface/acpidumps/commit/650b10a9264b7d82e0d70a7956ac1b1016dd3351.

**Comment by nickdepinet** [comment]
here's a fun one - to get secure boot working - either from LM21.2 Edge base or with the surface kernel - when you reboot to enroll mok boot will stick on windows logo before grub. if you spam enough you can get the following error to show
```
Failed to open \EFI\BOOT\mmx64.efi - Not Found
Failed to load image \EFI\BOOT\mmx64.efi - Not Found
Failed to start MokManager: Not Found
Something has gone seriously wrong: import_mok_state() failed
```
Re-inserting the liveusb and telling UEFI to boot from usb will pop MokManager and you can enroll your key, and booting will now work as normal. Very strange behavior.

**Comment by qzed** [comment]
Okay, I've added a `feature/slg3` branch to the [SAM repo](https://github.com/linux-surface/surface-aggregator-module/). Could you test that? E.g. run
- `git clone https://github.com/linux-surface/surface-aggregator-module`
- `cd surface-aggregator-module/module`
- `git checkout feature/slg3`
- `make clean all`
- `make modprobe-unload`
- `make insmod`

Ideally, the battery indicator should show up with that.

**Comment by nickdepinet** [comment]
After that, power statistics go from "unknown" to having proper details. I don't get a battery indicator in my tray, but I assume that has to do with doing this after boot.
![power-statistics](https://github.com/linux-surface/linux-surface/assets/2041026/b992ecb3-e87d-4a39-88e6-e1a658de1456)

**Comment by qzed** [comment]
Thanks! That looks like it's working. Can you also check if platform profiles work (e.g. use `surface profile`).

**Comment by nickdepinet** [comment]
Looks like its working:
![surface_profile](https://github.com/linux-surface/linux-surface/assets/2041026/d7a2da71-978e-43bd-87f6-bae877792f66)

**Comment by qzed** [comment]
New kernels for Fedora and Arch are currently compiling. Unfortunately, Ubuntu hasn't published a v6.5.8 tag yet, so Debian/Ubuntu packages will be a bit delayed.

**Comment by nickdepinet** [comment]
Thanks for your work on this! once the debian/ubuntu packages drop i'll run through the matrix and update the wiki

**Comment by willoma** [comment]
Hello,

I'm currently using Ubuntu 23.10, freshly installed on my freshly acquired Laptop Go 3.
I can confirm it works from branch feature/slg3, modules installed with `make dkms-install`.
I at least can see the battery level :)

In case I have problems, I will open other issues

**Comment by qzed** [comment]
FYI: I've merged the `feature/slg3` of the SAM repo into `master`. Still waiting on some new Ubuntu mainline tags for the Debian/Ubuntu package.

**Comment by qzed** [comment]
Thanks for checking up with them.

The timing is a bit unfortunate and highlights an issue with our setup: The fix for the v6.5 suspend problem landed upstream in v6.5.8. Before, we had it in our patches, but by rebasing onto v6.5.8 we've dropped it. Which means if we directly apply those patches to v6.5.7, the suspend fix will be missing. Conversely, if we would include it, `git am` would complain. And in our setup we don't have a good way to support multiple minor versions. Since it looks like it will take them some time though, I'll add the fix to the distro-specific patches for now.

**Comment by qzed** [comment]
Ubuntu/Debian packages are building now.

**Comment by qzed** [comment]
Should now be in the repo (and actually installable). Would be great if you could test them.

**Comment by willoma** [comment]
I have removed the module with `make dkms-uninstall`.
I have installed the kernel with the apt command provided by the documentation.
Then I have rebooted.

I can still see the battery level (meaning it seems to be loaded), and the suspend problem is gone.

All good so far.

**Comment by qzed** [comment]
Thanks!

**Comment by qzed** [comment]
I've added an entry to the feature table, would be great if you could check its correctness and fill in the missing items: https://github.com/linux-surface/linux-surface/wiki/Supported-Devices-and-Features.

**Comment by qzed** [comment]
I'll close this issue for now since it looks like base support is there. If anything specific does not work, please open a new issue for that.

**Comment by santialur** [comment]
> here's a fun one - to get secure boot working - either from LM21.2 Edge base or with the surface kernel - when you reboot to enroll mok boot will stick on windows logo before grub. if you spam enough you can get the following error to show
> 
> ```
> Failed to open \EFI\BOOT\mmx64.efi - Not Found
> Failed to load image \EFI\BOOT\mmx64.efi - Not Found
> Failed to start MokManager: Not Found
> Something has gone seriously wrong: import_mok_state() failed
> ```
> 
> Re-inserting the liveusb and telling UEFI to boot from usb will pop MokManager and you can enroll your key, and booting will now work as normal. Very strange behavior.

I am having this issue. I don't quite understand how to solve it. Would you mind giving me a hand?
